src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1188,7 +1188,7 @@ private async Task ProcessCompilationEventsAsync(AnalysisScope analysisScope, An
                     await ProcessEventAsync(completedEvent, analysisScope, analysisStateOpt, cancellationToken).ConfigureAwait(false);
                 }
             }
-            catch (Exception e) when (FatalError.ReportUnlessCanceled(e))
+            catch (Exception e) when (FatalError.ReportUnlessCanceled(e, cancellationToken))
```

Comment:
Seems like passing the `cancellationToken` to `ReportUnlessCanceled` will cause a semantic change. It now does an additional check `cancellationToken.IsCancellationRequested`, which did not happen previously. You may want to revert all changes to `ReportUnlessCanceled` callsites?

Reply:
yeah, I convinced myself that this was the right thing to do. I'll revert it here and create a separate PR where we can discuss if this change is what we want or not.

Reply:
reverted